sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+    elif method == "mean":
+        threshold = skimage.filters.threshold_mean(data)
+    elif method == "minimum":
+        threshold = skimage.filters.threshold_minimum(data)
+    elif method == "triangle":
+        threshold = skimage.filters.threshold_triangle(data)
+    elif method == "isodata":
+        threshold = skimage.filters.threshold_isodata(data)
+
+    return threshold
+
+
+def trim_intergranules(segmented_image, mark=False):
+    """
+    Remove the erronous idenfication of intergranule material in the middle of
+    granules that pure threshold segmentation produces.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect extra intergranules.
+    mark : `bool`
+        If False, remove erronous intergranules. If True, mark them as 0.5
+        instead (for later examination).
+
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndarray`
+        The segmented image without incorrect extra intergranules.
+    """
+
+    if len(np.unique(segmented_image)) > 2:
+        raise ValueError("segmented_image must have only values of 1 and 0")
+
+    segmented_image_fixed = np.copy(segmented_image).astype(float)
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    # find value of the large continuous 0-valued region
+    size = 0
+    for value in values:
+        if len((labeled_seg[labeled_seg == value])) > size:
+            real_IG_value = value
+            size = len(labeled_seg[labeled_seg == value])
+
+    # set all other 0 regions to mark value (1 or 0.5)
+    for value in values:
+        if np.sum(segmented_image[labeled_seg == value]) == 0:
+            if value != real_IG_value:
+                if not mark:
+                    segmented_image_fixed[labeled_seg == value] = 1
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+
+    return segmented_image_fixed
+
+
+def mark_faculae(segmented_image, data, res):
+    """
+    Mark faculae seperatley from granules - give them a value of 2 not 1.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect middles.
+    data : `numpy array`
+        The original flux values.
+    res : `float`
+        Spatial resolution (arcsec/pixel) of the data.
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndrray`
+        The segmented image with faculae marked as 1.5.
+    """
+
+    fac_size_limit = 2  # max size of a faculae in sqaure arcsec
+    fac_pix_limit = fac_size_limit / res
+    fac_brightness_limit = np.mean(data) + 0.5 * np.std(data)
+
+    if len(np.unique(segmented_image)) > 3:
+        raise ValueError("segmented_image must have only values of 1, 0, " + "an 0.5 (if dim centers marked)")
+
+    segmented_image = segmented_image
+    segmented_image_fixed = np.copy(segmented_image.astype(float))

```

Comment:
Does the float conversion matter?

Reply:
There was some reason why the region labeling wasn't working on in values. I am not sure why, but converting to float worked.

Reply:
Can you make a comment here for that please?